### DISCUSSION TITLE
publish MQTT topic on indi-allsky MQTT sensors chart

### DISCUSSION THREAD

**Discussion by quanticleap** [aaronwmorris/indi-allsky#1531] [Q&A] [answered]
Hi, the goal is graph on indi-allsky graph the data from a remote sensor through MQTT.

To simulate the publish of sensor data, I'm using an app on iphone, like mqttanalyser and easymqtt.

The mosquitto was installed on the indi-allsky with the install script and connection with username and password is fine from the app MQTTAnalyzer or Easymqtt, however the data published with topic like "sensor/temp" and payload: 24 is not graphed on indi-allsky.
On indi-allsky is configured the Config-Sensors-MQTT Broker Sensor and in Config-Admin-Extra lots-MQTT Broker Sensor C Value 1 ,2,3, 4 are configured to be on graph.

Once this will work, I will enable the real sensors and mqtt on esp32 .
When the app publish (topic and payload) to indi-allsky connected server, the data is passed to the server , I can see connections with tcpdump, but is not graphed.
The mosquitto is listening on 1883 tcp port and on netstat -na | grep 1883 , I can see the established connection with the iphone app.
1727168048: New connection from 192.168.1.8:63768 on port 1883.
1727168048: New client connected from 192.168.1.8:63768 as mqtt-analyzer-QOWhEJN3 (p5, c1, k60, u'indi-allsky')

Please , if you can advice how to correctly configure in order to receive and graph MQTT data on indi-allsky from a remote mqtt publisher sensor?

Thanks

**Comment by quanticleap** [comment]
I tried again with mqtt from esp32 with dht11 sensor and data arrives correctly on mqtt indi-allsky, but is not present on charts graph ( the password is changed in the next log.:

raspberry indi-allsky side:
pi@astro:~ $ mosquitto_sub -h localhost -t "esp32/dht11" -u "indi-allsky" -P "password"
Temp: 73.4 F / 23.0 C    Humidity: 17%
Temp: 73.4 F / 23.0 C    Humidity: 17%

and on esp32 side:
Published to esp32/dht11: Temp: 73.4 F / 23.0 C    Humidity: 17%
Temp: 73.4 F / 23.0 C    Humidity: 17% 
Published to esp32/dht11: Temp: 73.4 F / 23.0 C    Humidity: 17%
Temp: 73.4 F / 23.0 C    Humidity: 17% 
Published to esp32/dht11: Temp: 73.4 F / 23.0 C    Humidity: 17%

**Reply by quanticleap** [reply]
For the last test, I forgot to let only the integer. I will test again.

**Comment by quanticleap** [comment, marked as answer]
all good,the the MQTT data from esp32 with dht11 sensor is on the charts now.The data with two different topics to indi-allsky mqtt as "floats", one for temp and one for humidity and without other information, then on indi-allsky are configured both topics in "PIN" field of MQTT broker sensor, separated by ",".
Next the topics, the data received from CLI:
pi@astro:~ $ mosquitto_sub -h localhost -t "esp32/dht11/temp" -u "indi-allsky" -P "password"
23.00
23.00
pi@astro:~ $ mosquitto_sub -h localhost -t "esp32/dht11/hum" -u "indi-allsky" -P "password"
17.00
17.00

**Reply by aaronwmorris** [reply]
Fantastic!  Glad you got it worked out.

**Reply by quanticleap** [reply]
Thank you for this great software!😊
